src/RedisCache.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+use MediaWiki\MediaWikiServices;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Curse Inc.
+ * Redis Cache
+ *
+ * @author        Alexia E. Smith
+ * @copyright    (c) 2015 Curse Inc.
+ * @license        GNU General Public License v3.0 only
+ * @package        RedisCache
+ * @link        https://gitlab.com/hydrawiki
+ *
+ **/
+class RedisCache {
+	/**
+	 * Up and ready server connections.
+	 */
+	private array $servers = [];
+
+	/**
+	 * Last exception error.
+	 */
+	private string $lastError = '';
+
+	public function __construct( private array $redisServers, private LoggerInterface $logger ) { }
```

Comment:
**issue**: I'm not sure if the `redisServer` is always declared. I can't see default configuration in variables-base.php, and only those for prod/dev, so sandbox, preview, verify may throw TypeError exception.

I don't know why it was declared like that (maybe to not override cache on staging/testing environments), but there are options like prefix etc. which probably should help with that. Tho not sure if we should change that behaviour right now